lib/document.js

```diff
@@ -3897,6 +3899,24 @@ Document.prototype.populate = function populate() {
   return this;
 };
 
+/* Returns an array of all populated documents associated with the query. */
+Document.prototype.$getPopulatedDocs = function $getPopulatedDocs() {
+  const index = (Object.keys(this.$__.populated));
+  // console.log(index);
+  let result = [];
+
+  for (const key of index) {
+    let value = this.get(key);
+    if (Array.isArray(value)) result = result.concat(value);
```

Comment:
Mongoose style guidelines generally say no one-line ifs. Curly braces `{}` around `if` always, so this should be:

```javascript
if (Array.isArray(value)) {
  result = result.concat(value);
} else if (typeof value === 'object' && value !== null) {
  result.push(value);
}
```